library/src/com/google/maps/android/geojson/GeoJsonFeature.java

```diff
@@ -0,0 +1,211 @@
+package com.google.maps.android.geojson;
+
+import com.google.android.gms.maps.model.LatLngBounds;
+
+import java.util.HashMap;
+import java.util.Observable;
+import java.util.Observer;
+
+/**
+ * A GeoJsonFeature has a geometry, bounding box, id and set of properties. Styles are also stored
+ * in this class.
+ */
+public class GeoJsonFeature extends Observable implements Observer {
+
+    private final String mId;
+
+    private final LatLngBounds mBoundingBox;
+
+    private final HashMap<String, String> mProperties;
+
+    private GeoJsonGeometry mGeoJsonGeometry;
+
+    private GeoJsonPointStyle mGeoJsonPointStyle;
+
+    private GeoJsonLineStringStyle mGeoJsonLineStringStyle;
+
+    private GeoJsonPolygonStyle mGeoJsonPolygonStyle;
+
+    /**
+     * Creates a new GeoJsonFeature object
+     *
+     * @param GeoJsonGeometry type of geometry to assign to the feature
+     * @param id              id to refer to the feature by
+     * @param properties      hashmap of data containing properties related to the feature
+     * @param boundingBox     array defining the bounding box of the feature
+     */
+    public GeoJsonFeature(GeoJsonGeometry GeoJsonGeometry, String id,
+            HashMap<String, String> properties, LatLngBounds boundingBox) {
+        mGeoJsonGeometry = GeoJsonGeometry;
+        mId = id;
+        mBoundingBox = boundingBox;
+        mProperties = properties;
+        mGeoJsonPointStyle = new GeoJsonPointStyle();
+        mGeoJsonLineStringStyle = new GeoJsonLineStringStyle();
+        mGeoJsonPolygonStyle = new GeoJsonPolygonStyle();
+    }
+
+    /**
+     * Gets the hashmap of the properties
+     *
+     * @return hashmap of properties
+     */
+    public HashMap<String, String> getProperties() {
+        return mProperties;
+    }
+
+    /**
+     * Gets the GeoJsonPointStyle of the feature
+     *
+     * @return GeoJsonPointStyle object
+     */
+    public GeoJsonPointStyle getPointStyle() {
+        return mGeoJsonPointStyle;
+    }
+
+    /**
+     * Sets the GeoJsonPointStyle of the feature
+     *
+     * @param geoJsonPointStyle GeoJsonPointStyle object
+     */
+    public void setPointStyle(GeoJsonPointStyle geoJsonPointStyle) {
+        mGeoJsonPointStyle.deleteObserver(this);
+        mGeoJsonPointStyle = geoJsonPointStyle;
+        mGeoJsonPointStyle.addObserver(this);
+        checkRedrawFeature(mGeoJsonPointStyle);
+    }
+
+    /**
+     * Gets the GeoJsonLineStringStyle of the feature
```

Comment:
Doc comments in this format don't provide any more information than is already available from the method signature. Consider something like "Returns the style used when rendering LineStrings".